src/include/execution/sql/decimal_magic_numbers.h

```diff
@@ -0,0 +1,269 @@
+    {0x32b4bdfd4d668ecf, 0x825bb91604e810cc, 0x17c5be0019f60321, 0x28f1f9638c9fdf35},
+    {0xa2425ff75e14fc3, 0x1a1258379a94d028, 0xd18df2ccd1fe00a0, 0x3b6398471c1ff971},
+    {0x39d66589687f9e9, 0x01d59f290ee19dae, 0x8e31e14833001005, 0xf05c071c6998f1b3},
+    {0xcfb11ead453994ba, 0x67de18eda5814af2, 0x0b5b1aa028ccd99e, 0x59e338e387ad8e29},
+    {0x2989d2ef743eb758, 0x7b2c6b62bab37563, 0x9bdf05533b5c2b86, 0x11fa3e93e7ef82d5},
+    {0x42761e4bed31255a, 0x5ead789df785889f, 0x5fcb3bb85ef9df3c, 0xe990641fd97f37bb},
+    {0xd4ad2dbfc3d07787, 0x955e4ec64b44e864, 0x65bd8be79652ca5c, 0x85014065eb30b257},
+    {0x2a8909265a5ce4b4, 0xb77942f475742e7a, 0x7abf82618476f545, 0xb4337347957023ab},
+    {0x1039d428a8b8eaea, 0xfca1ac82efb45ca9, 0x77fcdc09b62c8824, 0x8149483089341779},
+    {0xb38fb9daa78e44ab, 0x2dcf7a6b19209442, 0x59949342bd140d07, 0x35420d1a7520258f}};
+
+/* Defines the 256 bit magic number division for powers of 10 with algo used and
+ * the constant p used during the division*/
+constexpr unsigned MAGIC_P_AND_ALGO_ARRAY[39][2] = {
+    {0, 0},   {259, 0}, {263, 1}, {266, 1}, {267, 0}, {272, 0}, {275, 0}, {279, 0}, {282, 0}, {285, 0},
+    {289, 0}, {291, 0}, {296, 1}, {299, 0}, {301, 0}, {305, 0}, {309, 0}, {313, 1}, {316, 1}, {320, 1},
+    {321, 0}, {325, 0}, {329, 0}, {333, 1}, {336, 1}, {339, 0}, {342, 0}, {346, 1}, {349, 0}, {350, 0},
+    {351, 0}, {359, 1}, {362, 0}, {363, 0}, {367, 0}, {372, 0}, {373, 0}, {379, 1}, {383, 1}};
+
+/** Magic Number for 128 bit division
+ * This struct is used to store Magic Numbers for 128 bit division
+ * It stores two 128 bit unsigned integers representing the upper and
+ * lower 64 bits of the magic number
+ * p Represents the adjust constant p during the magic number algorithm
+ * Algo is constant which is either 0 or 1 representing the type of algo
+ * we need for the division*/
+struct Magic_Number_128 {
+  /// Upper half of 128 bit magic number
+  uint128_t upper;
+  /// Lower half of 128 bit magic number
+  uint128_t lower;
+  /// value - p in magic division
+  unsigned p;
+  /// Algo type
+  unsigned algo;
+};
+
+/** Magic Number for 256 bit division
+ * This struct is used to store Magic Numbers for 256 bit division
+ * It stores 4 128 bit unsigned integers representing the magic number
+ * in 64 bits each.
+ * p Represents the adjust constant p during the magic number algorithm
+ * Algo is constant which is either 0 or 1 representing the type of algo
+ * we need for the division*/
+struct Magic_Number_256 {
+  /// Highest 64 bits
+  uint128_t A;
+  /// High Middle 64 bits
+  uint128_t B;
+  /// Low Middle 64 bits
+  uint128_t C;
+  /// Lowest 64 bits
+  uint128_t D;
+  /// value - p in magic division
+  unsigned p;
+  /// Algo type
+  unsigned algo;
+};
+
+/* Magic Array for 128 bit division with powers of 10
+ * This map is used after multiplication of decimals to get
+ * the correct result*/
+struct Magic_Number_128 MAGIC_MAP_128_BIT_POWER_TEN[39] = {{0, 0, 0, 0},
```

Comment:
1. Remove `struct`.
2. Comment format.
3. `constexpr` if you make the `Magic_Number_128` constructor `constexpr`, IIRC.